test/e2e/scheduling/limit_range.go

```diff
@@ -44,97 +44,34 @@ const (
-		selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
-
-		options := metav1.ListOptions{LabelSelector: selector.String()}
-		limitRanges, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).List(options)
-		framework.ExpectNoError(err, "failed to query for limitRanges")
-		framework.ExpectEqual(len(limitRanges.Items), 0)
-		options = metav1.ListOptions{
-			LabelSelector:   selector.String(),
-			ResourceVersion: limitRanges.ListMeta.ResourceVersion,
-		}
-
-		listCompleted := make(chan bool, 1)
-		lw := &cache.ListWatch{
-			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-				options.LabelSelector = selector.String()
-				limitRanges, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).List(options)
-				if err == nil {
-					select {
-					case listCompleted <- true:
-						e2elog.Logf("observed the limitRanges list")
-						return limitRanges, err
-					default:
-						e2elog.Logf("channel blocked")
-					}
-				}
-				return limitRanges, err
-			},
-			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-				options.LabelSelector = selector.String()
-				return f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Watch(options)
-			},
-		}
-		_, _, w, _ := watchtools.NewIndexerInformerWatcher(lw, &v1.LimitRange{})
-		defer w.Stop()
-
-		ginkgo.By("Submitting a LimitRange")
-		limitRange, err = f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Create(limitRange)
-		framework.ExpectNoError(err)
-
-		ginkgo.By("Verifying LimitRange creation was observed")
-		select {
-		case <-listCompleted:
-			select {
-			case event, _ := <-w.ResultChan():
-				if event.Type != watch.Added {
-					e2elog.Failf("Failed to observe limitRange creation : %v", event)
-				}
-			case <-time.After(e2eservice.RespondingTimeout):
-				e2elog.Failf("Timeout while waiting for LimitRange creation")
-			}
-		case <-time.After(e2eservice.RespondingTimeout):
-			e2elog.Failf("Timeout while waiting for LimitRange list complete")
-		}
+var (
+	f = framework.NewDefaultFramework("limitrange")
+)
 
+var _ = SIGDescribe("LimitRange", func() {
+	ginkgo.It("should create a LimitRange with defaults and ensure pod has those defaults applied", func() {
+		limitRange, expected := setupLimitRangeTest()
 		ginkgo.By("Fetching the LimitRange to ensure it has proper values")
-		limitRange, err = f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Get(limitRange.Name, metav1.GetOptions{})
+		limitRange, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Get(limitRange.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err)
-		expected := v1.ResourceRequirements{Requests: defaultRequest, Limits: defaultLimit}
 		actual := v1.ResourceRequirements{Requests: limitRange.Spec.Limits[0].DefaultRequest, Limits: limitRange.Spec.Limits[0].Default}
 		err = equalResourceRequirement(expected, actual)
 		framework.ExpectNoError(err)
+	})
 
+	ginkgo.It("should create a Pod with no resource requirements", func() {
```

Comment:
So I disagree that we should be breaking up this function into multiple It() blocks.  

It(should be the behavior you are verifying in the test)
By (whatever means necessary) 

^ It is ok to have multiple By()'s for a given test, but it's not ok to have multiple It()'s in a test if they are not germain to the the intent or subpackage.

Reply:
@timothysc Did you see @spiffxp's [comment about splitting this test into multiple tests](https://github.com/kubernetes/kubernetes/pull/78921#discussion_r295536235) due how busy it is?

Reply:
friendly ping @timothysc

Reply:
Actually, just noticed this other comment https://github.com/kubernetes/kubernetes/pull/78921#discussion_r298771876

Reply:
So you can break apart the test by cleaning up the behavior into clean functions and possibly collapsing the By's.  Comment above still holds.  Multiple It(s) is not the right approach.